emscripten.py

```diff
@@ -273,14 +273,15 @@ def save_settings():
         const = const[1:-1]
       const = '{ ' + const + ' }'
       args = []
-      arity = metadata['asmConstArities'][k]
+      arity = metadata['asmConstArities'][k][-1]
       for i in range(arity):
         args.append('$' + str(i))
       const = 'function(' + ', '.join(args ) + ') ' + const
       asm_consts[int(k)] = const
 
+    join = lambda outer: (item for inner in outer for item in inner)
```

Comment:
`join` might be confusing given `string.join`, how about calling it something more specific here? like `join_arities` or something better.